src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -581,4 +583,36 @@ private static void assertNotNull(AssertionInfo info, Class<?> actual) {
   private static void classParameterIsNotNull(Class<?> clazz) {
     requireNonNull(clazz, "The class to compare actual with should not be null");
   }
+
+  /**
+   * Verifies that the actual {@code Class} has the given {@code packagename}.
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param packageName the package that must be declared in the class.
+   */
+  public void assertHasPackage(AssertionInfo info, Class<?> actual, String packageName) {
+    assertNotNull(info, actual);
+    requireNonNull(packageName, shouldNotBeNull("package").create());
+    Package actualPackage = actual.getPackage();
+
+    if (actualPackage == null || !actualPackage.getName().equals(packageName)) {
+      throw failures.failure(info, shouldHavePackage(actual, packageName));
+    }
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the given {@code Package}.
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param aPackage the package that must be declared in the class.
+   */
+  public void assertHasPackage(AssertionInfo info, Class<?> actual, Package aPackage) {
+    assertNotNull(info, actual);
+    requireNonNull(aPackage, shouldNotBeNull("package").create());
+    Package actualPackage = actual.getPackage();
+
+    if (!java.util.Objects.equals(actualPackage, aPackage)) {
```

Comment:
As `aPackage` is non-null already at this point, this could be:

```suggestion
    if (!aPackage.equals(actualPackage)) {
```